src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -369,10 +368,20 @@ private static void LoadSourceFiles(CanvasDocument app, DirectoryReader director
 
                 // Json peer to a .pa file. 
                 var controlExtraData = file.ToObject<Dictionary<string, ControlState>>();
-                var topParentName = file._relativeName.Replace(".editorstate.json", "");
+                var topParentFilename = file._relativeName.Replace(".editorstate.json", "");
                 foreach (var control in controlExtraData)
                 {
-                    control.Value.TopParentName = Utilities.UnEscapeFilename(topParentName);
+                    // START v24 backwards compatibility
```

Comment:
Move to a function? `ApplyV24BackCompat(control)`?

Reply:
Good idea, I'll do that.